content/token22-in-the-client.md

```diff
@@ -0,0 +1,470 @@
+
+If we want to use the Token Extension Program when we call `getOrCreateAssociatedTokenAccount` Extension Tokens, we can pass in `TOKEN_2022_PROGRAM_ID` for the `tokenProgramId` parameter:
+
+```ts
+const tokenProgramId = TOKEN_2022_PROGRAM_ID;
+
+const tokenAccount = await getOrCreateAssociatedTokenAccount(
+  connection,
+  payer,
+  mintAddress,
+  payer.publicKey,
+  true,
+  'finalized',
+  {commitment: 'finalized'},
+  tokenProgramId // TOKEN_PROGRAM_ID for Token Program tokens and TOKEN_2022_PROGRAM_ID for Token Extension Program tokens
+)
+```
+
+To re-create the ATA's address from scratch, we can use the `findProgramAddressSync` function by providing the correct seeds:
+
+```ts
+function findAssociatedTokenAddress(
+  walletAddress: PublicKey,
+  tokenMintAddress: PublicKey
+): PublicKey {
+  return PublicKey.findProgramAddressSync(
+    [
+      walletAddress.toBuffer(),
+      TOKEN_PROGRAM_ID.toBuffer(), // replace TOKEN_PROGRAM_ID with TOKEN_2022_PROGRAM_ID for Token22 tokens
+      tokenMintAddress.toBuffer(),
+    ],
+    ASSOCIATED_TOKEN_PROGRAM_ID
+  )[0];
+}
+```
+
+## How to fetch tokens
+
+When it comes to fetching tokens, there is no difference between tokens and extension tokens. All we have to do is provide the correct token program:
+
+```ts
+const tokenAccounts = await connection.getTokenAccountsByOwner(
+	walletPublicKey,
+	{ programId: TOKEN_PROGRAM_ID } // or TOKEN_2022_PROGRAM_ID
+)
+```
+
+If we want to fetch all of the tokens for a particular owner, we can use a function like `getTokenAccountsByOwner`, and then call it twice, once with `TOKEN_PROGRAM_ID` and another with `TOKEN_2022_PROGRAM_ID`.
+
+```ts
+const allOwnedTokens = []
+const tokenAccounts = await connection.getTokenAccountsByOwner(
+	wallet.publicKey,
+	{programId: TOKEN_PROGRAM_ID}
+)
+const tokenExtensionAccounts = await connection.getTokenAccountsByOwner(
+	wallet.publicKey,
+	{programId: TOKEN_2022_PROGRAM_ID}
+)
+
+allOwnedTokens.push(...tokenAccounts, ...tokenExtensionAccounts)
+```
+
+NOTE: It may be advised to store and associate the token program with the token upon fetching.
+
+### Check owning program
+
+You may run into a scenario where you don't know the token program for a given account. Fortunately, `getParsedAccountInfo` will allow us to determine the owning program:
+
+```tsx
+const accountInfo = await connection.getParsedAccountInfo(mintAddress);
+if (accountInfo.value === null) {
+  throw new Error('Account not found');
+}
+
+const programId = accountInfo.value.owner; // will return TOKEN_PROGRAM_ID for Token Program mint address and TOKEN_2022_PROGRAM_ID for Token Extension Program mint address
+
+//we now use the programId to fetch the tokens
+const tokenAccounts = await connection.getTokenAccountsByOwner(
+  wallet.publicKey,
+  {programId}
+)
+```
+
+NOTE: After you fetch the owning account, it may be a good idea to save that owner and associate it with the mints/tokens you are handling.
```

Comment:
I like that you clarify 'owning program' in the headline (since we know people struggle with this) - maybe 'after you fetch the owning program's account' or 'after you fetch the owning program' to keep it consistent?